### ISSUE TITLE
Make chat commanders hide messages in chat

### ISSUE THREAD

**Issue by gaymeowing** [typical-developers/oaklands-feedback#969]
### Roblox Username

kalrnlo

### What do you think can have quality-of-life improvements?

chat commander

### How do you think things can be improved?

Currently when ever a player says a trigger word, the message is visible in chat. Ideally trigger words would be hidden in chat, by having a [ShouldDelieverCallback](https://create.roblox.com/docs/reference/engine/classes/TextChannel#ShouldDelieverCallback) registered to every chat channel under the TextChannels folder in TextChatService thats created at runtime.

code example because this api originally confused me with how its explained
```luau
local TextChatService = game:GetService("TextChatService")

-- using boolean? because its not guaranteed for a command to already be registered, and also using a set for O1 acess times because players can register any amount of commands
local PLAYER_COMMANDS = {} :: { [number]: { [string]: boolean? } }
local TEXT_CHANNELS: Folder = TextChatService:WaitForChild("TextChannels")  :: any

local function should_deliever(message: TextChatMessage): boolean
    local source = message.TextSource
    local is_command = PLAYER_COMMANDS[source.UserId][message.Text]

    return is_command == nil
end

TEXT_CHANNELS.ChildAdded:Connect(function(channel: TextChannel)
    channel.ShouldDelieverCallback = should_deliever
end :: any)

for _, channel in TEXT_CHANNELS:GetChildren() do
    channel.ShouldDelieverCallback = should_deliever
end
```

**Comment by gaymeowing** [comment]
Alternatively TextChatCommands could be used, but those block any message that includes an alias for the command. But could probably be worked around via messing with TextChatService.SendingMessage to then manually trigger TextChannel:SendAsync() if the player doesnt have that command registered.

**Comment by HooferDevelops** [comment]
I don't think this would work because commanders can be used by any player in the server (intentionally). I've been trying to think of a solution, my only idea is a new component but I'm not sure about that.

**Comment by gaymeowing** [comment]
> I don't think this would work because commanders can be used by any player in the server (intentionally). I've been trying to think of a solution, my only idea is a new component but I'm not sure about that.

Perhaps turn them into a remote spawner? That spawns a remote tool that can be used to trigger a command. With the ability to only allow the property owner to spawn a remote.
Remotes could be stylized to look like a big red button on a walkie talkie sorta object, with a antenna centered on the top.

Although this would go hand in hand with having a multi level hotbar like in satisfactory, and scrap mechanic. But that's another suggestion.

**Comment by HooferDevelops** [comment]
I like that idea a lot.
